### PR TITLE
Add Epomaker tide65

### PR DESCRIPTION
**Pull request by Epomaker**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/epomaker/tide65/EPOMAKER TIDE 65-fenlie.json
Comment (fauxpark) [inline keyboards/epomaker/tide65/EPOMAKER TIDE 65-fenlie.json:1]: VIA JSONs are not allowed. Please remove them.

File: keyboards/epomaker/tide65/EPOMAKER TIDE 65-long-space.json
Comment (fauxpark) [inline keyboards/epomaker/tide65/EPOMAKER TIDE 65-long-space.json:1]: As above.

File: keyboards/epomaker/tide65/config.h
Comment (fauxpark) [inline keyboards/epomaker/tide65/config.h:1]: Much of this file should be migrated to info.json.

File: keyboards/epomaker/tide65/epomaker_tide_65.c [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+#include QMK_KEYBOARD_H
+
+#ifdef RGB_MATRIX_ENABLE
+
+// clang-format off
+
+#undef ___
+#define ___ NO_LED
+
+led_config_t g_led_config = {
```
Comment (fauxpark) [inline keyboards/epomaker/tide65/epomaker_tide_65.c:26]: This should be migrated to info.json.

File: keyboards/epomaker/tide65/epomaker_tide_65.h [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+#pragma once
+
+#include "quantum.h"
+
+#define ___ KC_NO
+
+// clang-format off
+
+#define LAYOUT( \
```
Comment (fauxpark) [inline keyboards/epomaker/tide65/epomaker_tide_65.h:25]: This should be migrated to info.json.

File: keyboards/epomaker/tide65/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        KC_GRV,   KC_F1,      KC_F2,      KC_F3,    KC_F4,    KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,     KC_F11,   KC_F12,   _______,   _______,
+        _______,  _______,    _______,    _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,    KC_HOME,  KC_SCRL,  _______,   _______,
+        _______,  _______,    _______,    _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,    KC_PSCR,            _______,   _______,
+        _______,  _______,    _______,    _______,  _______,  _______,  _______,  _______,  _______,  _______,              _______,  _______,  _______,   _______,
+        _______,  GU_TOGG,    _______,    EE_CLR,   EE_CLR,   EE_CLR,   EE_CLR,                       _______,              _______,  _______,  _______,   _______
+        )
+}; 
+
+#ifdef ENCODER_MAP_ENABLE
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (fauxpark) [inline keyboards/epomaker/tide65/keymaps/via/keymap.c:27]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

File: keyboards/epomaker/tide65/readme.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+* Hardware Supported: EPOMAKER TIDE 65
+* Hardware Availability: [xxx](https://www.xxx.com)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make epomaker_tide_65:default
+
+Flashing example for this keyboard:
+
+    make epomaker_tide_65:default:flash
```
Comment (fauxpark) [inline keyboards/epomaker/tide65/readme.md:13]:
```suggestion
    make epomaker/tide65:default

Flashing example for this keyboard:

    make epomaker/tide65:default:flash
```

File: keyboards/epomaker/tide65/readme.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# EPOMAKER TIDE 65
+
+* Keyboard Maintainer: [sdk66](https://github.com/sdk66)
+* Hardware Supported: EPOMAKER TIDE 65
+* Hardware Availability: [xxx](https://www.xxx.com)
```
Comment (fauxpark) [inline keyboards/epomaker/tide65/readme.md:5]: Please fill this out.

File: keyboards/epomaker/tide65/rules.mk
Comment (fauxpark) [inline keyboards/epomaker/tide65/rules.mk:1]: Most of this file should be migrated to info.json.

**Review by fauxpark — CHANGES_REQUESTED** [review]
`info.json` is also missing completely...

File: keyboards/epomaker/tide65/readme.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# EPOMAKER TIDE 65
+
+* Keyboard Maintainer: [sdk66](https://github.com/sdk66)
+* Hardware Supported: EPOMAKER TIDE 65
+* Hardware Availability: [epomaker](https://www.epomaker.com)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make epomaker/epomaker_tide_65:default
```
Comment (fauxpark) [inline keyboards/epomaker/tide65/readme.md:9]:
```suggestion
    make epomaker/tide_65:default
```

File: keyboards/epomaker/tide65/keymaps/via/readme.md
Comment (fauxpark) [inline keyboards/epomaker/tide65/keymaps/via/readme.md:1]: This file can be deleted.

File: keyboards/epomaker/tide65/info.json [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+          {"matrix": [4, 9], "x": 10, "y": 4, "w": 1.25},
+          {"matrix": [4, 11], "x": 11.25, "y": 4, "w":1.25},
+          {"matrix": [4, 12], "x": 13, "y": 4},
+          {"matrix": [4, 13], "x": 14, "y": 4},
+          {"matrix": [4, 14], "x": 15, "y": 4}
+        ]
+      }
+    },
+    "rgb_matrix": {
+        "driver": "WS2812",
```
Comment (fauxpark) [inline keyboards/epomaker/tide65/info.json]:
```suggestion
        "driver": "ws2812",
```

File: keyboards/epomaker/tide65/info.json
Comment (fauxpark) [inline keyboards/epomaker/tide65/info.json:1]: Please run `qmk format-json -i` on this file.

File: keyboards/epomaker/tide65/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+        _______,  _______,    _______,    _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,    KC_PSCR,            _______,   _______,
+        _______,  _______,    _______,    _______,  _______,  _______,  _______,  _______,  _______,  _______,              _______,  _______,  _______,   _______,
+        _______,  GU_TOGG,    _______,    EE_CLR,   EE_CLR,   EE_CLR,   EE_CLR,                       _______,              _______,  _______,  _______,   _______
+        )
+};
+
+void keyboard_pre_init_user(void) {
+    setPinOutput(A5);
+    writePinHigh(A5);
+}
```
Comment (fauxpark) [inline keyboards/epomaker/tide65/keymaps/default/keymap.c:29]: What is this for?

File: keyboards/epomaker/tide65/info.json [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+    "usb": {
+      "vid": "0x342D",
+      "pid": "0xE463",
+      "suspend_wakeup_delay": 1000, 
+      "force_nkro": true, 
+      "device_version": "0.0.1"
+    },
+    "bootmagic": {
+      "matrix": [0, 0]
+    },
```
Comment (fauxpark) [inline keyboards/epomaker/tide65/info.json:15]:
```suggestion
    },
    "features": {
        "bootmagic": true,
        "mousekey": true,
        "extrakey": true,
        "nkro": true
    },
```

File: keyboards/epomaker/tide65/rules.mk [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+BOOTMAGIC_ENABLE = yes          # Enable Bootmagic Lite
+MOUSEKEY_ENABLE  = yes          # Mouse keys
+EXTRAKEY_ENABLE  = yes          # Audio control and System control
+CONSOLE_ENABLE   = no           # Console for debug
+COMMAND_ENABLE   = no           # Commands for debug and configuration
+NKRO_ENABLE      = yes          # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no           # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE  = no           # Enable keyboard RGB underglow
+HAPTIC_ENABLE    = no           # Haptic
+AUDIO_ENABLE     = no           # Audio output
```
Comment (fauxpark) [inline keyboards/epomaker/tide65/rules.mk:14]:
```suggestion
# This file intentionally left blank
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/epomaker/tide65/keymaps/via/readme.md
Comment (fauxpark) [inline keyboards/epomaker/tide65/keymaps/via/readme.md:1]: This file can be deleted.

File: keyboards/epomaker/tide65/keymaps/default/keymap.c (hunk 8 above)
Comment (Epomaker) [reply keyboards/epomaker/tide65/keymaps/default/keymap.c:29]: This is to trigger the LED pins; only when the voltage level is high, the backlights will be turned on.

**Comment by Epomaker** [comment]
@fauxpark 

I have completed the work on this Pull Request and am ready to merge it into the main branch. However, I have found that the merge operation is currently restricted, and only authorized users can perform it.

Could you please help me merge this Pull Request? If you have any questions or need further information, please feel free to let me know. Thank you!

File: keyboards/epomaker/tide65/keymaps/default/keymap.c (hunk 8 above)
Comment (fauxpark) [reply keyboards/epomaker/tide65/keymaps/default/keymap.c:29]: This should be done at keyboard level then.

File: keyboards/epomaker/tide65/info.json
Comment (fauxpark) [reply keyboards/epomaker/tide65/info.json:1]: This is still outstanding.

File: keyboards/epomaker/tide65/info.json [hunk 11]
```diff
@@ -0,0 +1,237 @@
+{
+    "manufacturer": "HS",
+    "keyboard_name": "EPOMAKER TIDE 65",
+    "maintainer": "sdk66",
+    "bootloader": "wb32-dfu",
+    "bootmagic": {
+        "matrix": [0, 0]
+    },
```
Comment (waffle87) [inline keyboards/epomaker/tide65/info.json:8]:
```suggestion
```

File: keyboards/epomaker/tide65/info.json [hunk 12]
```diff
@@ -0,0 +1,237 @@
...
+    "encoder": {
+        "rotary": [
+            {"pin_a": "B7", "pin_b": "B6"}
+        ]
+    },
+    "features": {
+        "bootmagic": true,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
```
Comment (waffle87) [inline keyboards/epomaker/tide65/info.json:19]: Probably want this here.
```suggestion
        "nkro": true,
        "rgb_matrix": true
```

File: keyboards/epomaker/tide65/info.json [hunk 13]
```diff
@@ -0,0 +1,237 @@
...
+            {"matrix": [4, 9], "x": 144, "y": 64, "flags": 4},
+            {"matrix": [4, 11], "x": 176, "y": 64, "flags": 4},
+            {"matrix": [4, 12], "x": 192, "y": 64, "flags": 4},
+            {"matrix": [4, 13], "x": 208, "y": 64, "flags": 4},
+            {"matrix": [4, 14], "x": 224, "y": 64, "flags": 4}
+        ],
+        "max_brightness": 100,
+        "sat_steps": 64,
+        "speed_steps": 51,
+        "val_steps": 25
```
Comment (waffle87) [inline keyboards/epomaker/tide65/info.json:148]: These values do not seem very user friendly. Is there reasoning behind them?

File: keyboards/epomaker/tide65/mcuconf.h [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+
+#pragma once
+
+#include_next <mcuconf.h>
+
+#undef WB32_SPI_USE_QSPI
+#define WB32_SPI_USE_QSPI TRUE
+
+#undef WB32_SPI_USE_SPIM2
+#define WB32_SPI_USE_SPIM2 TRUE
```
Comment (waffle87) [inline keyboards/epomaker/tide65/mcuconf.h:25]: What exactly is SPI used for on this keyboard?

File: keyboards/epomaker/tide65/config.h [hunk 15]
```diff
@@ -0,0 +1,9 @@
+// Copyright 2024 SDK (@sdk66)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+/* UART */
+#define SERIAL_DRIVER SD3
+#define SD1_TX_PIN C10
+#define SD1_RX_PIN C11
```
Comment (waffle87) [inline keyboards/epomaker/tide65/config.h:9]: As above; what purpose does this serve?

File: keyboards/epomaker/tide65/tide65.c [hunk 16]
```diff
@@ -0,0 +1,8 @@
+#include QMK_KEYBOARD_H
+
+// clang-format off
+void keyboard_pre_init_user(void) {
+    setPinOutput(A5);
+    writePinHigh(A5);
+}
+// clang-format on
```
Comment (waffle87) [inline keyboards/epomaker/tide65/tide65.c:8]:
```suggestion
// Copyright 2024 QMK
// SPDX-License-Identifier: GPL-2.0-or-later

#include "quantum.h"

void keyboard_pre_init_kb(void) {
    gpio_set_pin_output(A5);
    gpio_write_pin_high(A5);
    keyboard_pre_init_user();
}
```

File: keyboards/epomaker/tide65/halconf.h
Comment (waffle87) [inline keyboards/epomaker/tide65/halconf.h:1]: As above.

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/epomaker/tide65/info.json (hunk 11 above)
Comment (Epomaker) [reply keyboards/epomaker/tide65/info.json:8]: ill delete it

File: keyboards/epomaker/tide65/config.h
Comment (waffle87) [inline keyboards/epomaker/tide65/config.h:1]: This file should be removed, as it does not contain anything.

File: keyboards/epomaker/tide65/halconf.h
Comment (waffle87) [inline keyboards/epomaker/tide65/halconf.h:1]: This file should be removed, as it does not contain anything.

File: keyboards/epomaker/tide65/mcuconf.h
Comment (waffle87) [inline keyboards/epomaker/tide65/mcuconf.h:1]: This file should be removed, as it does not contain anything.

File: keyboards/epomaker/tide65/tide65.c [hunk 17]
```diff
@@ -0,0 +1,14 @@
+// Copyright 2024 SDK (@sdk66)
+// SPDX-License-Identifier: GPL-2.0-or-later
+#include QMK_KEYBOARD_H
```
Comment (waffle87) [inline keyboards/epomaker/tide65/tide65.c:3]:
```suggestion
#include "quantum.h"
```

File: keyboards/epomaker/tide65/tide65.c [hunk 18]
```diff
@@ -0,0 +1,14 @@
...
+
+// clang-format off
+void keyboard_pre_init_kb(void) {
+    setPinOutput(A5);
+    writePinHigh(A5);
```
Comment (waffle87) [inline keyboards/epomaker/tide65/tide65.c:8]:
```suggestion
    gpio_set_pin_output(A5);
    gpio_write_pin_high(A5);
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/epomaker/tide65/info.json [hunk 19]
```diff
@@ -0,0 +1,232 @@
...
+        "force_nkro": true,
+        "pid": "0xE463",
+        "suspend_wakeup_delay": 1000,
+        "vid": "0x342D"
+    },
+    "ws2812": {
+        "pin": "B15"
+    },
+    "layouts": {
+        "LAYOUT_ortho_5x15": {
```
Comment (zvecr) [inline keyboards/epomaker/tide65/info.json:157]:
```suggestion
        "LAYOUT": {
```

File: keyboards/epomaker/tide65/info.json [hunk 20]
```diff
@@ -0,0 +1,232 @@
...
+            "jellybean_raindrops": true,
+            "multisplash": true,
+            "pixel_flow": true,
+            "pixel_fractal": true,
+            "pixel_rain": true,
+            "rainbow_beacon": true,
+            "rainbow_moving_chevron": true,
+            "rainbow_pinwheels": true,
+            "raindrops": true,
+            "solid_color": true,
```
Comment (zvecr) [inline keyboards/epomaker/tide65/info.json:58]:
```suggestion
```

File: keyboards/epomaker/tide65/keymaps/default/keymap.c [hunk 21]
```diff
@@ -0,0 +1,26 @@
+// Copyright 2024 SDK (@sdk66)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+// clang-format off
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+    [0] = LAYOUT_ortho_5x15( /* Base */
```
Comment (zvecr) [inline keyboards/epomaker/tide65/keymaps/default/keymap.c:9]:
```suggestion
    [0] = LAYOUT( /* Base */
```

File: keyboards/epomaker/tide65/keymaps/default/keymap.c [hunk 22]
```diff
@@ -0,0 +1,26 @@
...
+        KC_ESC,   KC_1,       KC_2,       KC_3,     KC_4,     KC_5,     KC_6,     KC_7,     KC_8,     KC_9,     KC_0,       KC_MINS,  KC_EQL,   KC_BSPC,   KC_MUTE,
+        KC_TAB,   KC_Q,       KC_W,       KC_E,     KC_R,     KC_T,     KC_Y,     KC_U,     KC_I,     KC_O,     KC_P,       KC_LBRC,  KC_RBRC,  KC_BSLS,   KC_DEL,
+        KC_CAPS,  KC_A,       KC_S,       KC_D,     KC_F,     KC_G,     KC_H,     KC_J,     KC_K,     KC_L,     KC_SCLN,    KC_QUOT,            KC_ENT,    KC_PGUP,  
+        KC_LSFT,  KC_Z,       KC_X,       KC_C,     KC_V,     KC_B,     KC_N,     KC_M,     KC_COMM,  KC_DOT,               KC_SLSH,  KC_RSFT,  KC_UP,     KC_PGDN,
+        KC_LCTL,  KC_LGUI,    KC_LALT,    KC_SPC,   KC_SPC,   KC_SPC,   KC_SPC,                       KC_RALT,              MO(1),    KC_LEFT,  KC_DOWN,   KC_RGHT
+        ),  
+
+    [1] = LAYOUT_ortho_5x15( /* Base */
```
Comment (zvecr) [inline keyboards/epomaker/tide65/keymaps/default/keymap.c:17]:
```suggestion
    [1] = LAYOUT(
```

File: keyboards/epomaker/tide65/keymaps/via/keymap.c [hunk 23]
```diff
@@ -0,0 +1,33 @@
+// Copyright 2024 SDK (@sdk66)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+// clang-format off
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+    [0] = LAYOUT_ortho_5x15( /* Base */
```
Comment (zvecr) [inline keyboards/epomaker/tide65/keymaps/via/keymap.c:9]:
```suggestion
    [0] = LAYOUT( /* Base */
```

File: keyboards/epomaker/tide65/keymaps/via/keymap.c [hunk 24]
```diff
@@ -0,0 +1,33 @@
...
+        KC_ESC,   KC_1,       KC_2,       KC_3,     KC_4,     KC_5,     KC_6,     KC_7,     KC_8,     KC_9,     KC_0,       KC_MINS,  KC_EQL,   KC_BSPC,   KC_MUTE,
+        KC_TAB,   KC_Q,       KC_W,       KC_E,     KC_R,     KC_T,     KC_Y,     KC_U,     KC_I,     KC_O,     KC_P,       KC_LBRC,  KC_RBRC,  KC_BSLS,   KC_DEL,
+        KC_CAPS,  KC_A,       KC_S,       KC_D,     KC_F,     KC_G,     KC_H,     KC_J,     KC_K,     KC_L,     KC_SCLN,    KC_QUOT,            KC_ENT,    KC_PGUP,  
+        KC_LSFT,  KC_Z,       KC_X,       KC_C,     KC_V,     KC_B,     KC_N,     KC_M,     KC_COMM,  KC_DOT,               KC_SLSH,  KC_RSFT,  KC_UP,     KC_PGDN,
+        KC_LCTL,  KC_LGUI,    KC_LALT,    KC_SPC,   KC_SPC,   KC_SPC,   KC_SPC,                       KC_RALT,              MO(1),    KC_LEFT,  KC_DOWN,   KC_RGHT
+        ),  
+
+    [1] = LAYOUT_ortho_5x15( /* Base */
```
Comment (zvecr) [inline keyboards/epomaker/tide65/keymaps/via/keymap.c:17]:
```suggestion
    [1] = LAYOUT( /* Base */
```

File: keyboards/epomaker/tide65/keymaps/via/keymap.c [hunk 25]
```diff
@@ -0,0 +1,33 @@
...
+        _______,  _______,    _______,    _______,  _______,  _______,  _______,  _______,  _______,  _______,  _______,    KC_PSCR,            _______,   _______,
+        _______,  _______,    _______,    _______,  _______,  _______,  _______,  _______,  _______,  _______,              _______,  _______,  _______,   _______,
+        _______,  GU_TOGG,    _______,    EE_CLR,   EE_CLR,   EE_CLR,   EE_CLR,                       _______,              _______,  _______,  _______,   _______
+        )
+};
+
+#ifdef ENCODER_MAP_ENABLE
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [0] = {ENCODER_CCW_CW(KC_VOLD, KC_VOLU)},
+    [1] = {ENCODER_CCW_CW(KC_VOLD, KC_VOLU)},
```
Comment (zvecr) [inline keyboards/epomaker/tide65/keymaps/via/keymap.c:29]:
```suggestion
    [1] = {ENCODER_CCW_CW(_______, _______)},
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by Epomaker** [comment]
@zvecr 
I have completed the work on this Pull Request and am ready to merge it into the main branch. However, I have found that the merge operation is currently restricted, and only authorized users can perform it.

Could you please help me merge this Pull Request? If you have any questions or need further information, please feel free to let me know. Thank you!

**Review by drashna — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**